sycl/unittests/helpers/PiMockPlugin.hpp

```diff
@@ -36,6 +37,13 @@ template <class T> inline T createDummyHandle(size_t Size = 0) {
   return reinterpret_cast<T>(DummyHandlePtr);
 }
 
+// Allocates a dummy handle of type T with support of reference counting
+// and associates it with the provided Data.
+template <class T> inline T createDummyHandleWithData(unsigned char *Data) {
+  DummyHandlePtrT DummyHandlePtr = new DummyHandleT(Data);
+  return reinterpret_cast<T>(DummyHandlePtr);
```

Comment:
Why doesn't it violate ANSI-alias rules?

Reply:
This is a helper function for creating underlying data structure for native handles in the mock plugin (e.g. pi_mem). Like with proper plugins, these handles are never dereferenced directly and only used inside mock plugin's PI functions after reinterpret casting back to DummyHandlePtrT.